perfkitbenchmarker/linux_benchmarks/terasort_benchmark.py

```diff
@@ -0,0 +1,136 @@
+"""
+
+import datetime
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import spark_service
+from perfkitbenchmarker import flags
+
+
+
+BENCHMARK_NAME = 'terasort_benchmark'
+BENCHMARK_CONFIG = """
+terasort_benchmark:
+  description: Run the Apache hadoop terasort benchmark on a cluster.
+  spark_service:
+    service_type: managed
+    num_workers: 4
+"""
+
+# This points to a file on the spark cluster.
+TERASORT_JARFILE = ('file:///usr/lib/hadoop-mapreduce/'
+                    'hadoop-mapreduce-examples.jar')
+TERAGEN_CLASSNAME = 'org.apache.hadoop.examples.terasort.TeraGen'
+TERASORT_CLASSNAME = 'org.apache.hadoop.examples.terasort.TeraSort'
+TERAVALIDATE_CLASSNAME = 'org.apache.hadoop.examples.terasort.TeraValidate'
+
+flags.DEFINE_integer('terasort_dataset_size', 10000,
+                     'Data set size to generate')
+flags.DEFINE_string('terasort_unsorted_dir', 'tera_gen_data', 'Location of '
+                    'the unsorted data. TeraGen writes here, and TeraSort '
+                    'reads from here.')
+
+flags.DEFINE_string('terasort_sorted_dir', 'tera_sort_dir', 'Location for the '
+                    'sorted data. TeraSort writes to here, TeraValidate reads '
+                    'from here.')
+flags.DEFINE_string('terasort_validate_dir', 'tera_validate_dir', 'Output of '
+                    'the TeraValidate command')
+
+FLAGS = flags.FLAGS
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def Prepare(benchmark_spec):
+  pass
+
+
+def Run(benchmark_spec):
+  """Executes the given jar on the specified Spark cluster.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  spark_cluster = benchmark_spec.spark_service
+  start = datetime.datetime.now()
+
+  results = []
+  metadata = spark_cluster.GetMetadata()
+  gen_args = [str(FLAGS.terasort_dataset_size), FLAGS.terasort_unsorted_dir]
+  sort_args = [FLAGS.terasort_unsorted_dir, FLAGS.terasort_sorted_dir]
+  validate_args = [FLAGS.terasort_sorted_dir, FLAGS.terasort_validate_dir]
+
+  stages = [('generate', TERAGEN_CLASSNAME, gen_args),
+            ('sort', TERASORT_CLASSNAME, sort_args),
+            ('validate', TERAVALIDATE_CLASSNAME, validate_args)]
+  for (label, classname, args) in stages:
+    stats = spark_cluster.SubmitJob(TERASORT_JARFILE,
+                                    classname,
+                                    job_type=spark_service.HADOOP_JOB_TYPE,
+                                    job_arguments=args)
+    if not stats[spark_service.SUCCESS]:
+      raise Exception('Stage {0} unsuccessful'.format(label))
+    current_time = datetime.datetime.now()
+    results.append(sample.Sample(label + '_wall_time',
+                                 (current_time - start).total_seconds(),
```

Comment:
Another style thing but to match up with the rest of the code can you do this?
```wall_time = (current_time - start).total_seconds()```
and then use that?  It makes it easier to read as all the others do not have calculations in the constructor